### DISCUSSION TITLE
fastrtps-shared-targets.cmake contains absolute path

### DISCUSSION THREAD

**Discussion by kjeldflarup** [eProsima/Fast-DDS#5082] [Q&A]
I'm including Fast-DDS in a cross compiled and embedded system, and providing it into a SDK, using the Yocto distribution.

Now in the SDK I get a fastrtps-shared-targets.cmake with this line

```
  INTERFACE_LINK_LIBRARIES "fastcdr;foonathan_memory;-lpthread;dl;/tmp/yocto/tmp-glibc/work/cortexa53-crypto-oe-linux/fast-dds/2.14.2-r0/recipe-sysroot/usr/lib/libtinyxml2.so;\$<\$<BOOL:1>:OpenSSL::SSL\$<SEMICOLON>OpenSSL::Crypto\$<\$<BOOL:>:\$<SEMICOLON>crypt32.lib>>;\$<\$<BOOL:>:iphlpapi\$<SEMICOLON>Shlwapi>;-lpthread;rt"
```

This contains the path of libtinyxml2.so at the compile time in the Yocto recipe, which is not valid any where else.
I have searched and searched, for some options to enforce using just `-ltinyxml2`

Any hints to how I force the build to do this, please.

**Comment by JesusPoderoso** [comment]
Hi @kjeldflarup, thanks for using Fast DDS, and sorry for the inconvenience.
There is a way to force the build, by using the [thirdparty library options](https://fast-dds.docs.eprosima.com/en/latest/installation/configuration/cmake_options.html#third-party-libraries-options).
Please include the CMake argument `-DTHIRDPARTY_TinyXML2=FORCE` during the build process to use the internal library.
